pkg/thing/interactors/unregister_thing.go

```diff
@@ -0,0 +1,40 @@
+package interactors
+
+import (
+	"errors"
+)
+
+// Unregister runs the use case to remove a registered thing
+func (i *ThingInteractor) Unregister(authorization, id string) error {
+	i.logger.Debug("Executing unregister thing use case")
+
+	if authorization == "" {
+		return errors.New("authorization key not provided")
+	}
+
+	if id == "" {
+		return errors.New("thing's id not provided")
+	}
+
+	err := i.thingProxy.RemoveThing(authorization, id)
+	if err != nil {
+		msg := err.Error()
+		sendErr := i.clientPublisher.SendUnregisterDevice(id, &msg)
+		if sendErr != nil {
+			return sendErr
+		}
+		return err
+	}
+
+	sendErr := i.connectorPublisher.SendUnregisterDevice(id)
+	if sendErr != nil {
+		i.logger.Error(sendErr)
```

Comment:
Shouldn't the error be returned?

Reply:
@joaoaneto In this case, return the function at this point will avoid sending a client request response. However the error at sending a connector request should be treated right. By now I add a TODO at this code point to analyse carefully the possible errors and what should be done.

Reply:
Ok, ack.